sympy/solvers/solveset.py

```diff
@@ -1473,8 +1462,11 @@ def add_type(eq, symbol, domain, flags):
         elif _is_exponential(simplified_equation, symbol):
             new_eq = _solve_expo(simplified_equation, symbol)
         # check if it is logarithmic type equation
-        elif _is_logarithmic(simplified_equation, symbol):
-            new_eq = _solve_log(simplified_equation, symbol)
+        elif _is_logarithm_reducable_to_single_instance(
+                Eq(lhs, rhs), symbol):
```

Comment:
Passing equation as an instance of `Eq`, so that in the helpers above there will not be a need to call `invert_real` again to get the `lhs` and `rhs`.